### PR TITLE
Support file id in content fragment

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces support for `fileId` in the private API for posting content fragments. When a `fileId` is provided, it will be associated with the content fragment, allowing it to be used during conversation rendering for the model.

Key changes:
- Add an optional `fileId` column to the `content_fragments` table
- Create an index on `fileId`
- Allow either `fileId` or `content`/`contentType` when posting a content fragment via the private API

Future work:
- Use the `fileId` for rendering conversations for the model.

## Risk

Worst case it breaks the content fragment logic.The file api is not yet used in production. Safe to rollback.

## Deploy Plan

It requires to run a migration:
```
psql $FRONT_DATABASE_URI -f ./migrations/db/migration_32.sql
```

### CODE REVIEW

File: front/lib/api/assistant/conversation/content_fragment.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      fileModelId: file.id,
+      sourceUrl,
+      textBytes: null,
+      title,
+    });
+  }
+}
+
+// Injects a new content fragment in the conversation.
+export async function postNewContentFragment(
```
Comment (spolu) [inline front/lib/api/assistant/conversation/content_fragment.ts:93]: can we keep this one in conversations since it has the ranking logic and such :pray:

Comment (spolu) [reply front/lib/api/assistant/conversation/content_fragment.ts:93]: and really this is an operation on content fragment

**Review by spolu — APPROVED** [review]
Looks very reasonable

File: front/lib/api/assistant/conversation/content_fragment.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/api/assistant/conversation/content_fragment.ts]: Done.

File: front/lib/models/workspace.ts [hunk 2]
```diff
@@ -224,7 +224,7 @@ MembershipInvitation.init(
     indexes: [
       { fields: ["workspaceId", "status"] },
       { unique: true, fields: ["sId"] },
-      { fields: ["email", "status"] },
+      { fields: ["inviteEmail", "status"] },
```
Comment (flvndvd) [inline front/lib/models/workspace.ts:227]: 🚗

File: front/migrations/db/migration_31.sql [hunk 3]
```diff
@@ -1,2 +1,3 @@
 -- Migration created on Jul 03, 2024
-CREATE INDEX CONCURRENTLY "membership_invitations_email_status" ON "membership_invitations" ("email", "status");
+CREATE INDEX CONCURRENTLY "membership_invitations_invite_email_status" ON "membership_invitations" ("inviteEmail", "status");
```
Comment (flvndvd) [inline front/migrations/db/migration_31.sql:2]: 🚗

**Pull request merged**